packages/graphql-toolbox/src/modules/SchemaView/Favorites/FavoriteEntry.tsx

```diff
@@ -0,0 +1,130 @@
+    updateName: (newName: string, id: string) => void;
+    dragHandle: JSX.Element;
+}
+
+export const FavoriteEntry = ({ dragHandle, onSelectFavorite, favorite, updateName }: FavoriteEntryProps) => {
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const [nameValue, setNameValue] = useState<string>(favorite.name);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const finishEditingName = () => {
+        setEditMode(false);
+
+        // Ensure empty strings are reverted to previous name value
+        if (nameValue === "") {
+            setNameValue(favorite.name);
+            return;
+        }
+
+        updateName(nameValue, favorite.id);
+    };
+
+    const _handleKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+        if (event.key === "Enter") {
+            finishEditingName();
+        }
+    };
+
+    useEffect(() => {
+        if (editMode && inputRef.current) {
+            inputRef.current.focus();
+            inputRef.current.setSelectionRange(0, inputRef.current.value.length);
+        }
+    }, [editMode]);
+
+    return (
+        <li
+            className={
+                "favorite-entry flex justify-between items-center p-1 mb-1 cursor-pointer hover:n-bg-neutral-40 rounded"
+            }
+        >
+            <div className="w-full flex items-center justify-start">
+                <div className="hover-show">{dragHandle}</div>
+                {editMode ? (
+                    <TextInput
+                        className="w-60"
+                        value={nameValue}
+                        ref={inputRef}
+                        onChange={(event) => setNameValue(event.currentTarget.value)}
+                        onKeyDown={_handleKeyDown}
+                        size="extra-small"
+                        aria-label="Input for new name for the favorite snippet"
+                    />
+                ) : (
+                    <Checkbox
+                        onChange={(event) => {
+                            if (event.target.checked) {
+                                useFavoritesStore.getState().addSelectedFavorite(favorite.id);
+                            } else {
+                                useFavoritesStore.getState().removeSelectedFavorite(favorite.id);
+                            }
+                        }}
+                        checked={useFavoritesStore.getState().selectedFavorites.includes(favorite.id)}
+                        className="w-full"
+                        label={favorite.name}
+                    />
+                )}
+            </div>
+
+            <IconButton
+                aria-label="Delete favorite"
+                clean
+                onClick={() => onSelectFavorite(favorite.typeDefs)}
+                onKeyDown={() => onSelectFavorite(favorite.typeDefs)}
+                className="hover-show"
+            >
+                <PlayCircleIconOutline color={"#006FD6"} />
```

Comment:
I think it would be good to use `tokens` here from `import { tokens } from "@neo4j-ndl/base";`. Don't know the value which is needed for this case here.

```suggestion
                <PlayCircleIconOutline color={tokens.colors.primary[50]} />
```